src/optimum/nvidia/builder/base.py

```diff
@@ -461,31 +461,25 @@ def _build_engine_for_rank(
         )
 
         builder = Builder()
+
+        # TODO: handling this with kwargs is ugly.
+        builder_config_kwargs = self.get_builder_config_kwargs(config=config, qconfig=qconfig, shard=shard, is_parallel=is_parallel, opt_level=opt_level)
+
+        # TODO: Why are we using `fp8` here while TRT-LLM examples are using `int8`?
+        # TODO: Only TP=1 is supported for Whisper? it is hardcoded in TensorRT-LLM/examples/whisper/run.py
         build_config = builder.create_builder_config(
             name=config["model_type"],
             precision=self._dtype.value,
             fp8=qconfig.mode.has_fp8_qdq(),
-            vocab_size=config.vocab_size,
             hidden_size=config.hidden_size,
-            hidden_act=config.activation,
-            num_heads=config.num_heads,
-            num_kv_heads=config.num_kv_heads,
             num_layers=config.num_layers,
-            max_position_embeddings=config.max_sequence_length,
             max_batch_size=self._optimization_profile.max_batch_size,
-            max_input_len=self._optimization_profile.max_prompt_length,
-            max_output_len=self._optimization_profile.max_output_length,
-            max_num_tokens=None,
-            max_beam_width=self._beam_width,
-            strongly_typed=qconfig.mode.has_fp8_qdq(),
             tensor_parallel=shard.tp_size,
-            pipeline_parallel=shard.pp_size,
-            parallel_build=is_parallel,
             use_refit=False,
             quant_mode=self._quantization_config.mode,
-            opt_level=opt_level,
```

Comment:
Here we remove the model-specific builder config arguments from the `base.py` file that is shared by several models.